editoast/src/views/path/path_item_cache.rs

```diff
@@ -51,17 +56,30 @@ impl PathItemCache {
             .map(|part| (infra_id, part.track.0.clone()))
             .chain(tracks);
         let existing_track_ids =
-            TrackSectionModel::retrieve_batch_unchecked::<_, Vec<_>>(conn, tracks)
+            TrackSectionModel::retrieve_batch_unchecked::<_, Vec<_>>(conn, tracks.clone())
                 .await?
                 .into_iter()
                 .map(|track| track.obj_id)
                 .collect();
+        let existing_track_labels =
+            TrackSectionModel::retrieve_batch_unchecked::<_, Vec<_>>(conn, tracks)
```

Comment:
These make 2 identical query to the database. Maybe it's better to store the result and reuse it?

```rust
let track_section_models = TrackSectionModel::retrieve_batch_unchecked::<_, Vec<_>>(conn, tracks.clone()).await?;
let existing_track_ids = track_section_models.iter()...
let existing_track_labels = track_section_models.iter()...
```